src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -0,0 +1,478 @@
+                    # the one that we are adding.
+                    new_name = table.next_available_name(
+                        old_sym.name, other_table=routine_table)
+                    routine_table.rename_symbol(old_sym, new_name)
+                    table.add(old_sym)
+
+            # Check whether this symbol is used to specify the precision of
+            # any literals.
+            if old_name in precision_map:
+                for lit in precision_map[old_name]:
+                    # A literal is immutable so create a new one with the
+                    # updated symbol as its precision.
+                    dtype = ScalarType(lit.datatype.intrinsic, old_sym)
+                    lit.replace_with(Literal(lit.value, dtype))
+
+    def validate(self, node, options=None):
+        '''
+        Checks that the supplied node is a valid target for inlining.
+
+        Routines with a return value (AKA functions in Fortran) are currently
+        rejected: TODO #924.
+
+        :param node: target PSyIR node.
+        :type node: subclass of :py:class:`psyclone.psyir.nodes.Routine`
+        :param options: a dictionary with options for transformations.
+        :type options: dict of str:values or None
+
+        :raises TransformationError: if the supplied node is not a Call.
+        :raises TransformationError: if the routine has a return value.
+        :raises TransformationError: if the routine body contains a Return \
+            that is not the first or last statement.
+        :raises TransformationError: if the routine body contains a CodeBlock.
+        :raises TransformationError: if the called routine has a named \
+            argument.
+        :raises TransformationError: if a symbol of a given name is imported \
+            from different containers at the call site and within the routine.
+        :raises TransformationError: if the routine accesses an un-resolved \
+            symbol.
+        :raises TransformationError: if a symbol declared in the parent \
+            container is accessed in the target routine.
+        :raises TransformationError: if any of the actual arguments represent \
+            an array subsection.
+        :raises TransformationError: if the shape of an array dummy argument \
+            does not match that of the corresponding actual argument.
+
+        '''
+        super().validate(node, options=options)
+
+        # The node should be a Call.
+        if not isinstance(node, Call):
+            raise TransformationError(
+                f"The target of the InlineTrans transformation "
+                f"should be a Call but found '{type(node).__name__}'.")
+
+        name = node.routine.name
+
+        # Check that we can find the source of the routine being inlined.
+        routine = self._find_routine(node)
+
+        if not routine.children or isinstance(routine.children[0], Return):
+            # An empty routine is fine.
+            return
```

Comment:
Can you move this after the check for return_symbol immediately below. Otherwise a function with only a return may be considered an empty routine.

Reply:
Oops - good spot.